Failed to replace StorageClass on GKE 1.24

When running the e2e tests, after creating a GKE cluster, we add labels in the GCE storage classes using `kubectl replace --force -f`.

https://github.com/elastic/cloud-on-k8s/blob/aee78413cf50e3b96f45d9aba4565968c7c93c02/hack/deployer/runner/gke.go#L169-L209

This step failed on k8s 1.24: 
- [cloud-on-k8s-e2e-tests-gke-k8s-versions/846/pipeline](https://devops-ci.elastic.co/blue/organizations/jenkins/cloud-on-k8s-e2e-tests-gke-k8s-versions/detail/cloud-on-k8s-e2e-tests-gke-k8s-versions/846/pipeline)
- [cloud-on-k8s-e2e-tests-gke-k8s-versions/845/pipeline](https://devops-ci.elastic.co/blue/organizations/jenkins/cloud-on-k8s-e2e-tests-gke-k8s-versions/detail/cloud-on-k8s-e2e-tests-gke-k8s-versions/845/pipeline)
- [cloud-on-k8s-e2e-tests-gke-k8s-versions/842/pipeline](https://devops-ci.elastic.co/blue/organizations/jenkins/cloud-on-k8s-e2e-tests-gke-k8s-versions/detail/cloud-on-k8s-e2e-tests-gke-k8s-versions/842/pipeline)

```
[2022-10-12T00:57:19.346Z] storageclass.storage.k8s.io "premium-rwo" deleted
[2022-10-12T01:02:19.704Z] error: timed out waiting for the condition
[2022-10-12T01:02:22.010Z] storageclass.storage.k8s.io "premium-rwo" deleted
[2022-10-12T01:07:22.410Z] error: timed out waiting for the condition
[2022-10-12T01:07:25.118Z] storageclass.storage.k8s.io "premium-rwo" deleted
[2022-10-12T01:12:25.410Z] error: timed out waiting for the condition
[2022-10-12T01:12:26.313Z] storageclass.storage.k8s.io "premium-rwo" deleted
[2022-10-12T01:17:26.716Z] error: timed out waiting for the condition
[2022-10-12T01:17:29.725Z] storageclass.storage.k8s.io "premium-rwo" deleted
[2022-10-12T01:22:30.157Z] error: timed out waiting for the condition
[2022-10-12T01:22:30.158Z] Error: exit status 1
[2022-10-12T01:22:30.158Z] exit status 1
[2022-10-12T01:22:30.158Z] make: *** [Makefile:534: run-deployer] Error 1
```

Normal log:
```
storageclass.storage.k8s.io "premium-rwo" deleted                                                                                                                                                                                     
storageclass.storage.k8s.io/premium-rwo replaced                                                                                                                                   
```

Relates to #5949, #5955.

--- a/hack/deployer/runner/gke.go
+++ b/hack/deployer/runner/gke.go
@@ -186,7 +186,7 @@ func (d *GKEDriver) setupLabelsForGCEProvider() error {
 		// This is a GCE storage class patch it
 
 		// start by removing the label that makes the storage class managed by the addon manager to prevent it from being recreated before us
-		err := exec.NewCommand(fmt.Sprintf("kubectl label sc %s \"addonmanager.kubernetes.io/mode\"-", storageClass.Name)).WithoutStreaming().Run()
+		err := exec.NewCommand(fmt.Sprintf(`kubectl label sc %s "addonmanager.kubernetes.io/mode"-`, storageClass.Name)).WithoutStreaming().Run()
 		if err != nil {
 			return err
 		}

--- a/hack/deployer/runner/gke.go
+++ b/hack/deployer/runner/gke.go
@@ -184,6 +184,13 @@ func (d *GKEDriver) setupLabelsForGCEProvider() error {
 			continue
 		}
 		// This is a GCE storage class patch it
+
+		// start by removing the label that makes the storage class managed by the addon manager to prevent it from being recreated before us
+		err := exec.NewCommand(fmt.Sprintf("kubectl label sc %s \"addonmanager.kubernetes.io/mode\"-", storageClass.Name)).WithoutStreaming().Run()
+		if err != nil {
+			return err
+		}
+
 		if storageClass.Parameters == nil {
 			storageClass.Parameters = make(map[string]string)
 		}

